app/src/main/java/com/example/extrememe/services/DatabaseDataLoader.java

```diff
@@ -17,4 +20,12 @@ public static FirebaseFirestore getDB() {
 
         return fireStoreDB;
     }
+
+    public static StorageReference getStorageRef(String url) {
+        if(storageRef == null) {
+            storageRef = storage.getReferenceFromUrl(url);
```

Comment:
the storage object might be null - android is known for static object get deliminated during lifecycle
https://stackoverflow.com/questions/9541688/static-variable-null-when-returning-to-the-app

- [ ]  FirebaseStorage.getInstance().getReferenceFromUrl(url)